assets/out

```diff
@@ -208,14 +242,61 @@ helm_upgrade() {
       $helm_bin diff "${helm_diff_args[@]}" | tee "$logfile"
     fi
   fi
-
+  helm_exit_status=0
   if [ "$exit_after_diff" = true ]; then
     echo "Exiting after diff"
   else
-    echo "Running command helm ${helm_echo_args[@]} | tee $logfile"
-    $helm_bin "${helm_args[@]}" $tls_flag | tee "$logfile"
+    echo "Running command helm ${helm_echo_args[@]}"
+    $helm_bin "${helm_args[@]}" $tls_flag > "$logfile" &
+    # note that we have to remove the tee in the above line because if we don't the helm_pid will be the pid of the tee process
+    # we are running both $helm_bin and helm_log_upgrade in the background so we can see the logs of the helm upgrade
+    # in near real-time while also tailing the logs of the pods in the namespace
+    helm_pid=$!
+    tail -f "$logfile" &
+    tail_pid=$!
+    echo "helm_pid: $helm_pid"
+    helm_log_upgrade $helm_pid &
+    helm_log_pid=$!
+    wait $helm_pid
+    helm_exit_status=$?
+    kill -9 $tail_pid 
```

Comment:
`kill -9` seems harsh. If there's buffered output, it won't display. Is the default `SIGTERM` not working?